File: c-sharp/ApiTests/PremPostHandlerTest.cs
using System.Net.Http;
using System.Threading.Tasks;
using FluentAssertions;
using Npb.Agview.Api.Example;
using Xunit;

namespace Npb.Agview.ApiTests.Example
{
    public class PremPostHandlerTest
    {
        private readonly PremPostHandler _sut;

        private readonly HttpClient _httpClient = new();
        private readonly string BaseUrl = Constants.BaseUrl;
        private readonly string ApiKey = Constants.ApiKey;
        private readonly string ApiSecret = Constants.ApiSecret;

        public PremPostHandlerTest()
        {
            var accessTokenHandler = new AccessTokenHandler(_httpClient, BaseUrl, ApiKey, ApiSecret);
            var premDbHandler = new PremDbHandler(Constants.PremsFilePath, Constants.PremAddressesFilePath);

            _sut = new PremPostHandler(_httpClient, accessTokenHandler, BaseUrl, premDbHandler);
        }

        [Fact]
        public async Task CreatesPrems()
        {
            var actual = await _sut.CreatePrems();

            actual.Should().HaveCountGreaterThan(0);
        }
    }
}

File: java/src/test/java/org/npb/agview/api/example/PremPostHandlerTest.java
package org.npb.agview.api.example;

import com.fasterxml.jackson.core.JsonProcessingException;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.net.http.HttpClient;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.*;
import static org.npb.agview.api.example.Constants.*;

class PremPostHandlerTest {

    private PremPostHandler sut;

    private final String BASE_URL = System.getenv("NPB_BASE_URL");
    private final String API_KEY = System.getenv("NPB_API_KEY");
    private final String API_SECRET = System.getenv("NPB_API_SECRET");
    private final HttpClient httpClient = HttpClient.newHttpClient();

    @BeforeEach
    public void setup() {
        var dbHandler = new PremDbHandler(DB_DIRECTORY+"/prem.csv",
                DB_DIRECTORY+"/prem_address.csv", new DbHandler());
        var connectionInfo = new Arguments(BASE_URL, API_KEY, API_SECRET);
        var accessTokenHandler = new AccessTokenHandler(httpClient, connectionInfo);
        sut = new PremPostHandler(httpClient,
                connectionInfo,
                accessTokenHandler,
                dbHandler);
    }

    @Test
    public void createsPrems() throws JsonProcessingException {
        var actual = sut.createPrems();

        assertThat(actual, hasSize(greaterThan(0)));
    }

}
